app/services/openai_provider.py

```
@@ -0,0 +1,111 @@
+                ],
+                temperature=0.3,
+                max_tokens=2000
+            )
+            
+            translation = response.choices[0].message.content.strip()
+            if not translation:
+                raise ProviderError(self.name, "Empty translation response")
+            return translation
+            
+        except openai.AuthenticationError as e:
+            raise ProviderError(self.name, f"Authentication failed: {str(e)}", e)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)